File: 이상욱/BOJ/m2w9/BOJ21921.java
package m2w9;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ21921 {
	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int n = Integer.parseInt(st.nextToken());
		int x = Integer.parseInt(st.nextToken());
		
		int[] arr = new int[n];
		
		st = new StringTokenizer(br.readLine());
		
		for(int i = 0; i < n; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		
		long sum = 0L;
		
		for(int i = 0; i < x; i++) {
			sum+=arr[i];
		}
		int cnt = 1;
		long max = sum;
		
		for(int i = x; i < n; i++) {
			sum+=arr[i];
			sum-=arr[i-x];
			
			if(sum > max) {
				max = sum;
				cnt = 1;
			}else if(sum == max) {
				cnt++;
			}
		}
		
		if(max == 0L) {
			System.out.println("SAD");
		}else {
			System.out.println(max);
			System.out.println(cnt);
		}
		
	}
}

File: 이상욱/BOJ/m2w9/BOJ21921.kt
package m2w9

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))
    var st = StringTokenizer(br.readLine())

    val n = st.nextToken().toInt()
    val x = st.nextToken().toInt()

    val arr = IntArray(n)

    st = StringTokenizer(br.readLine())

    for(i in 0 until n){
        arr[i] = st.nextToken().toInt()
    }

    var sum = 0L

    for(i in 0 until x){
        sum+=arr[i]
    }
    var cnt = 1
    var max = sum
    for(i in x until n){
        sum+=arr[i]
        sum-=arr[i-x]
        if(sum > max){
            max = sum
            cnt = 1
        }else if(sum == max){
            cnt++
        }
    }

    if(max == 0L){
        println("SAD")
    }else {
        println(max)
        println(cnt)
    }

}